src/globals.c

```diff
@@ -7,8 +7,8 @@
 
 /* Global objects */
 
-block_t *BLOCKS;
-int blocks_idx = 0;
+block_t *BLOCKS_HEAD;
+block_t *BLOCKS_TAIL;
```

Comment:
Rename to `BLOCKS_head` and `BLOCKS_tail`.

Reply:
Is it better to use another defined structure `block_info_t` that contains both member `head`, `tail`?

Reply:
Seems so, this keeps global variables tidy, I will later address this.